namespaces/exec.go

```diff
@@ -156,7 +156,7 @@ func killAllPids(container *libcontainer.Config) error {
 		freeze  = fs.Freeze
 		getPids = fs.GetPids
 	)
-	if systemd.UseSystemd() {
+	if container.Cgroups.Driver == "systemd" && systemd.UseSystemd() {
```

Comment:
I think we should remove the `systemd.UseSystemd()` call here.  This can be called by the caller, i.e. Docker, to determine what value to set as the Cgroups.Driver.  

If you set the driver to systemd and you don't have systemd enabled then it should fail hard because it's the callers fault and we cannot honor what they requested.